### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by jackluo-zz**
various fixes: add @throws tag, chang DOMString to String, add more sample codes in localStorage, use @constructedBy in database, change @desc to @description

### CODE REVIEW

**Comment by jeffheifetz** [comment]
Can you please update your descriptions to start with a capital and end with a period?

File: api/html5_database.js [hunk 1]
```diff
@@ -98,8 +100,8 @@ Database ={
  */
 SQLTransaction = {
 		/**
-		 * @desc This method executes the provided SQL statement
-		 * @param {DOMString} sqlStatement the SQL statement
+		 * @description This method executes the provided SQL statement
+		 * @param {String} sqlStatement the SQL statement
 		 * @param {ObjectArray} [arguments] the optional arguments used to preprocess a SQL statement by binding each ? placeholder with the value of the argument in the arguments array with the same position.
```
Comment (jeffheifetz) [inline api/html5_database.js]: Shouldn't this be {Object[]}

**Comment by jackluo-zz** [comment]
Nice catch. I will change it. thanks. We can use Array type for it.

**Comment by jeffheifetz** [comment]
Looks great

**Pull request merged**